packages/ckeditor5-ui/src/bindings/draggableviewmixin.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module TODO
+ */
+
+import type View from '../view';
+import { type Constructor, type Mixed, global, type EventInfo } from '@ckeditor/ckeditor5-utils';
+
+/**
+ * TODO
+ */
+export default function DraggableViewMixin<Base extends Constructor<View>>( view: Base ): Mixed<Base, DraggableView>
+{
+	abstract class DraggableMixin extends view {
+		/**
+		 * TODO
+		 */
+		declare public isDraggable: boolean;
+
+		/**
+		 * TODO
+		 */
+		private _onDragBound = this._onDrag.bind( this );
+
+		/**
+		 * TODO
+		 */
+		private _onDragEndBound = this._onDragEnd.bind( this );
+
+		/**
+		 * TODO
+		 */
+		private _isDragging: boolean = false;
+
+		/**
+		 * TODO
+		 */
+		private _currentCoordinates: { x: number; y: number } = { x: 0, y: 0 };
+
+		/**
+		 * TODO
+		 */
+		constructor( ...args: Array<any> ) {
+			super( ...args );
+
+			this.set( 'isDraggable', true );
+
+			if ( this.isRendered ) {
+				this._attachListeners();
+			} else {
+				this.on( 'render', () => {
+					this._attachListeners();
+				} );
+			}
+		}
+
+		/**
+		 * TODO
+		 */
+		private _attachListeners() {
+			this.listenTo( this.element!, 'mousedown', this._onDragStart.bind( this ) );
+			this.listenTo( this.element!, 'touchstart', this._onDragStart.bind( this ) );
+		}
+
+		/**
+		 * TODO
+		 */
+		private _attachDragDurationListeners() {
+			this.listenTo( global.document, 'mouseup', this._onDragEndBound );
+			this.listenTo( global.document, 'touchend', this._onDragEndBound );
+			this.listenTo( global.document, 'mousemove', this._onDragBound );
+			this.listenTo( global.document, 'touchmove', this._onDragBound );
+		}
+
+		/**
+		 * TODO
+		 */
+		private _detachDragDurationListeners() {
+			this.stopListening( global.document, 'mouseup', this._onDragEndBound );
+			this.stopListening( global.document, 'touchend', this._onDragEndBound );
+			this.stopListening( global.document, 'mousemove', this._onDragBound );
+			this.stopListening( global.document, 'touchmove', this._onDragBound );
+		}
+
+		/**
+		 * TODO
+		 */
+		private _onDragStart( evt: EventInfo, domEvt: MouseEvent | TouchEvent ) {
+			if ( !this.isDraggable || !this._isHandleEventTarget( domEvt ) ) {
+				return;
+			}
+
+			this._attachDragDurationListeners();
+
+			const x = ( domEvt instanceof TouchEvent ? domEvt.touches[ 0 ] : domEvt ).clientX;
+			const y = ( domEvt instanceof TouchEvent ? domEvt.touches[ 0 ] : domEvt ).clientY;
+
+			this._currentCoordinates = { x, y };
+
+			this._isDragging = true;
```

Comment:
It would be useful if we e.g. had some CSS bound to the dragging process, but I don't see such plans so it seems we can safely remove it and restore later if necessary.

Reply:
Hmm.. good call.. It could be useful for "outside world" or for some custom behaviors/styling. But in this case, it should be an observable public property, right? I am not sure what is our approach to mixed-in properties, i.e. if we specify them as private or public if we expect them to be used by the host class.

Reply:
I think it does make sense to make it a public observable. I don't see why that could be an issue for now :thinking:

Reply:
Okay, so let's make that a public observable.